File: Data Structures & Algorithms/concatenation-of-array/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    getConcatenation(nums) {
        const length = nums.length;
        const ansArr = new Array(length * 2)

        for (let i = 0; i < length; i++) {
            ansArr[i] = nums[i]
            ansArr[i + length] = nums[i]
        }

        return ansArr
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        L, R = 0, len(nums) - 1

        while L < R:
            M = (R + L) // 2
            if M == L or M == R: break
            if nums[L] < nums[M] and nums[M] < nums[R]:
                return nums[L]
            if nums[L] < nums[M]:
                L = M
            else:
                R = M
                
        return min(nums[L], nums[R])